cdap-app-templates/cdap-etl/cdap-etl-realtime/src/test/java/co/cask/cdap/template/etl/realtime/RealtimeElasticsearchSinkTest.java

```diff
@@ -0,0 +1,164 @@
+    node = nodeBuilder().settings(elasticsearchSettings.build()).client(false).node();
+    client = node.client();
+  }
+
+  @After
+  public void afterTest() {
+    node.close();
+  }
+
+  @BeforeClass
+  public static void setupTests() throws IOException {
+    // TODO: should only deploy test source and Elasticsearch sink
+    addTemplatePlugins(TEMPLATE_ID, "realtime-sources-1.0.0.jar",
+                       DataGeneratorSource.class, JmsSource.class, KafkaSource.class,
+                       TwitterSource.class, SqsSource.class);
+    addTemplatePlugins(TEMPLATE_ID, "realtime-sinks-1.0.0.jar",
+                       RealtimeCubeSink.class, RealtimeTableSink.class,
+                       StreamSink.class, RealtimeElasticsearchSink.class);
+    addTemplatePlugins(TEMPLATE_ID, "transforms-1.0.0.jar",
+                       ProjectionTransform.class, ScriptFilterTransform.class,
+                       StructuredRecordToGenericRecordTransform.class);
+    deployTemplate(NAMESPACE, TEMPLATE_ID, ETLRealtimeTemplate.class,
+                   PipelineConfigurable.class.getPackage().getName(),
+                   ETLStage.class.getPackage().getName(),
+                   RealtimeSource.class.getPackage().getName());
+  }
+
+  @Test
+  @Category(SlowTests.class)
+  public void testESSink() throws Exception {
+    ETLStage source = new ETLStage("DataGenerator", ImmutableMap.of(DataGeneratorSource.PROPERTY_TYPE,
+                                                                    DataGeneratorSource.TABLE_TYPE));
+    try {
+      ETLStage sink = new ETLStage("Elasticsearch",
+                                   ImmutableMap.of(Properties.Elasticsearch.TRANSPORT_ADDRESSES,
+                                                   InetAddress.getLocalHost().getHostName() + ":9300",
+                                                   Properties.Elasticsearch.CLUSTER, "testcluster",
+                                                   Properties.Elasticsearch.INDEX_NAME, "test",
+                                                   Properties.Elasticsearch.TYPE_NAME, "testing",
+                                                   Properties.Elasticsearch.ID_FIELD, "name"
+                                   ));
+      List<ETLStage> transforms = new ArrayList<>();
+      ETLRealtimeConfig etlConfig = new ETLRealtimeConfig(source, sink, transforms);
+      Id.Adapter adapterId = Id.Adapter.from(NAMESPACE, "esSinkTest");
+      AdapterConfig adapterConfig = new AdapterConfig("", TEMPLATE_ID.getId(), GSON.toJsonTree(etlConfig));
+      AdapterManager adapterManager = createAdapter(adapterId, adapterConfig);
+
+      adapterManager.start();
+      TimeUnit.SECONDS.sleep(5);
+      adapterManager.stop();
+
+      SearchResponse searchResponse = client.prepareSearch("Bob").execute().actionGet();
+      Assert.assertEquals(1, searchResponse.getHits().getTotalHits());
+
+      Assert.assertEquals(1, (int) searchResponse.getHits().getAt(0).field("id").getValue());
```

Comment:
maybe pull searchResponse.getHits().getAt(0) out into a variable since it's used alot in the block below